deployment/docker/Makefile

```diff
@@ -32,7 +32,7 @@ app-via-docker: package-base package-builder
 		--env GO111MODULE=on \
 		$(DOCKER_OPTS) \
 		$(BUILDER_IMAGE) \
-		go build $(RACE) -mod=vendor -trimpath -ldflags "-s -w -extldflags '-static' $(GO_BUILDINFO)" -tags 'netgo osusergo' \
+		go build $(RACE) -mod=vendor -trimpath -ldflags "-extldflags '-static' $(GO_BUILDINFO)" -tags 'netgo osusergo' \
```

Comment:
Unfortunately this increases binary size :(

As I know, [CPU and memory profiling](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/README.md#profiling) works without debug symbols.

@faceair , it would be great if you could provide examples when debug symbols are necessary?

Reply:
I often use `perf` to check the performance of programs on my linux machine.

For example `perf top`, when keep debug symbols:
![image](https://user-images.githubusercontent.com/3659110/80427785-0937de00-891b-11ea-83e9-aec2283e68d4.png)

without debug symbols:
![image](https://user-images.githubusercontent.com/3659110/80427895-3f755d80-891b-11ea-98fe-407c75a05d06.png)

Reply:
Also I remember that pprof doesn't support cgo very well, some cgo calls are not visible, is that right ?

Reply:
> I often use perf to check the performance of programs on my linux machine.

This is valid case. Note that debug symbols are stripped only from `-prod` binaries built with `make *-prod` command. These binaries are published then in Docker images and [binary releases](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). When you run `make *` without `-prod` suffix (for example, `make victoria-metrics`), then the resulting binary contains all the debugging symbols. So, probably, it would be better building VictoriaMetrics binaries from sources according to [these docs](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/README.md#how-to-build-from-sources) when you need debug symbols in `perf`.

> Also I remember that pprof doesn't support cgo very well, some cgo calls are not visible, is that right ?

I didn't notice this.

Reply:
I mean pprof can only see all the time consumed by cgocall, not every call therein.
![image](https://user-images.githubusercontent.com/3659110/80676007-646afb80-8ae8-11ea-8e7e-5c3e697fd673.png)

But perf can see the overhead of such things as ZSTD_compressBlock_doubleFast, ZSTD_decompressSequences_bmi2.
![image](https://user-images.githubusercontent.com/3659110/80676078-8d8b8c00-8ae8-11ea-827a-b6922a4261e5.png)

If you think it's more important to keep it smaller you can close this pull request, I can build VictoriaMetrics binaries myself

Reply:
Interesting. Never knew about this! Anyway, I think that `zstd` library is already highly tuned, so it will be hard to optimize it further :)
I usually use `go tool pprof` for profiling VictoriaMetrics and never used `perf` for this.

`go tool pprof` allows identifying particular source code lines responsible for high CPU usage (with `list <functioName>` command). As I understood, `perf` shows only per-function stats.

Additionally, `go tool pprof` allows collecting and analyzing other profiles additionally to CPU profiles:
- memory allocation and memory usage profiles (they are useful for reducing memory usage or CPU usage spent on GC) - see https://www.freecodecamp.org/news/how-i-investigated-memory-leaks-in-go-using-pprof-on-a-large-codebase-4bec4325e192/ .
- block and mutex contention profiles (useful for debugging concurrency bottlenecks) - see https://rakyll.org/mutexprofile/ .

Reply:
Hi @valyala, sorry for the late reply.

If the VM is performance tuning ready in the production environment, I think it would be better to keep the debug symbols, which help us to some extent to understand the running state of the program better.

It doesn't seem to add extra cost, although increasing the program size a little (about 4m), as a program running on the server side this extra size is nothing.

I understand that you are demanding a lot of neatness in your code (https://github.com/VictoriaMetrics/VictoriaMetrics#community-and-contributions), but keeping debug symbols does not add additional junk code, which I think is consistent with KISS principles.

Reply:
Agreed. I'm going to merge this pull request